### ISSUE TITLE
ResolvePackageAssets will fail with an exception when run concurrently

### ISSUE THREAD

**Issue by ericstj** [dotnet/sdk#2188]
This was introduced with a recent SDK we picked up in CoreFx.

During testing we generate projects and concurrently call a test target in that project to test asset resolution under different RIDs.

Granted what we are doing is a little sketchy, but RAR handles it better so I thought I'd file a bug.

When RAR can't open its cache file it merely emits a warning, however ResolvePackageAssets will fail with a stack trace:
```
D:\j\workspace\windows-TGrou---0d2c9ac4\bin\testPkg\tools\sdk\2.1.300-preview2-008530\Sdks\Microsoft.NET.Sdk\targets\Microsoft.PackageDependencyResolution.targets(205,5): error MSB4018: The "ResolvePackageAssets" task failed unexpectedly. [D:\j\workspace\windows-TGrou---0d2c9ac4\bin\testPkg\projects\System.Composition.TypedParts\netstandard1.7\project.csproj] [D:\j\workspace\windows-TGrou---0d2c9ac4\pkg\test\testPackages.proj]
D:\j\workspace\windows-TGrou---0d2c9ac4\bin\testPkg\tools\sdk\2.1.300-preview2-008530\Sdks\Microsoft.NET.Sdk\targets\Microsoft.PackageDependencyResolution.targets(205,5): error MSB4018: System.IO.IOException: The process cannot access the file 'D:\j\workspace\windows-TGrou---0d2c9ac4\bin\testPkg\projects\System.Composition.TypedParts\netstandard1.7\obj\Debug\netstandard1.7\project.assets.cache' because it is being used by another process. [D:\j\workspace\windows-TGrou---0d2c9ac4\bin\testPkg\projects\System.Composition.TypedParts\netstandard1.7\project.csproj] [D:\j\workspace\windows-TGrou---0d2c9ac4\pkg\test\testPackages.proj]
D:\j\workspace\windows-TGrou---0d2c9ac4\bin\testPkg\tools\sdk\2.1.300-preview2-008530\Sdks\Microsoft.NET.Sdk\targets\Microsoft.PackageDependencyResolution.targets(205,5): error MSB4018: at System.IO.FileStream.ValidateFileHandle(SafeFileHandle fileHandle) [D:\j\workspace\windows-TGrou---0d2c9ac4\bin\testPkg\projects\System.Composition.TypedParts\netstandard1.7\project.csproj] [D:\j\workspace\windows-TGrou---0d2c9ac4\pkg\test\testPackages.proj]
D:\j\workspace\windows-TGrou---0d2c9ac4\bin\testPkg\tools\sdk\2.1.300-preview2-008530\Sdks\Microsoft.NET.Sdk\targets\Microsoft.PackageDependencyResolution.targets(205,5): error MSB4018: at System.IO.FileStream.CreateFileOpenHandle(FileMode mode, FileShare share, FileOptions options) [D:\j\workspace\windows-TGrou---0d2c9ac4\bin\testPkg\projects\System.Composition.TypedParts\netstandard1.7\project.csproj] [D:\j\workspace\windows-TGrou---0d2c9ac4\pkg\test\testPackages.proj]
D:\j\workspace\windows-TGrou---0d2c9ac4\bin\testPkg\tools\sdk\2.1.300-preview2-008530\Sdks\Microsoft.NET.Sdk\targets\Microsoft.PackageDependencyResolution.targets(205,5): error MSB4018: at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options) [D:\j\workspace\windows-TGrou---0d2c9ac4\bin\testPkg\projects\System.Composition.TypedParts\netstandard1.7\project.csproj] [D:\j\workspace\windows-TGrou---0d2c9ac4\pkg\test\testPackages.proj]
D:\j\workspace\windows-TGrou---0d2c9ac4\bin\testPkg\tools\sdk\2.1.300-preview2-008530\Sdks\Microsoft.NET.Sdk\targets\Microsoft.PackageDependencyResolution.targets(205,5): error MSB4018: at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share) [D:\j\workspace\windows-TGrou---0d2c9ac4\bin\testPkg\projects\System.Composition.TypedParts\netstandard1.7\project.csproj] [D:\j\workspace\windows-TGrou---0d2c9ac4\pkg\test\testPackages.proj]
D:\j\workspace\windows-TGrou---0d2c9ac4\bin\testPkg\tools\sdk\2.1.300-preview2-008530\Sdks\Microsoft.NET.Sdk\targets\Microsoft.PackageDependencyResolution.targets(205,5): error MSB4018: at System.IO.File.Open(String path, FileMode mode, FileAccess access, FileShare share) [D:\j\workspace\windows-TGrou---0d2c9ac4\bin\testPkg\projects\System.Composition.TypedParts\netstandard1.7\project.csproj] [D:\j\workspace\windows-TGrou---0d2c9ac4\pkg\test\testPackages.proj]
D:\j\workspace\windows-TGrou---0d2c9ac4\bin\testPkg\tools\sdk\2.1.300-preview2-008530\Sdks\Microsoft.NET.Sdk\targets\Microsoft.PackageDependencyResolution.targets(205,5): error MSB4018: at Microsoft.NET.Build.Tasks.ResolvePackageAssets.CacheWriter..ctor(ResolvePackageAssets task) [D:\j\workspace\windows-TGrou---0d2c9ac4\bin\testPkg\projects\System.Composition.TypedParts\netstandard1.7\project.csproj] [D:\j\workspace\windows-TGrou---0d2c9ac4\pkg\test\testPackages.proj]
D:\j\workspace\windows-TGrou---0d2c9ac4\bin\testPkg\tools\sdk\2.1.300-preview2-008530\Sdks\Microsoft.NET.Sdk\targets\Microsoft.PackageDependencyResolution.targets(205,5): error MSB4018: at Microsoft.NET.Build.Tasks.ResolvePackageAssets.CacheReader..ctor(ResolvePackageAssets task) [D:\j\workspace\windows-TGrou---0d2c9ac4\bin\testPkg\projects\System.Composition.TypedParts\netstandard1.7\project.csproj] [D:\j\workspace\windows-TGrou---0d2c9ac4\pkg\test\testPackages.proj]
D:\j\workspace\windows-TGrou---0d2c9ac4\bin\testPkg\tools\sdk\2.1.300-preview2-008530\Sdks\Microsoft.NET.Sdk\targets\Microsoft.PackageDependencyResolution.targets(205,5): error MSB4018: at Microsoft.NET.Build.Tasks.ResolvePackageAssets.ReadItemGroups() [D:\j\workspace\windows-TGrou---0d2c9ac4\bin\testPkg\projects\System.Composition.TypedParts\netstandard1.7\project.csproj] [D:\j\workspace\windows-TGrou---0d2c9ac4\pkg\test\testPackages.proj]
D:\j\workspace\windows-TGrou---0d2c9ac4\bin\testPkg\tools\sdk\2.1.300-preview2-008530\Sdks\Microsoft.NET.Sdk\targets\Microsoft.PackageDependencyResolution.targets(205,5): error MSB4018: at Microsoft.NET.Build.Tasks.ResolvePackageAssets.ExecuteCore() [D:\j\workspace\windows-TGrou---0d2c9ac4\bin\testPkg\projects\System.Composition.TypedParts\netstandard1.7\project.csproj] [D:\j\workspace\windows-TGrou---0d2c9ac4\pkg\test\testPackages.proj]
D:\j\workspace\windows-TGrou---0d2c9ac4\bin\testPkg\tools\sdk\2.1.300-preview2-008530\Sdks\Microsoft.NET.Sdk\targets\Microsoft.PackageDependencyResolution.targets(205,5): error MSB4018: at Microsoft.NET.Build.Tasks.TaskBase.Execute() [D:\j\workspace\windows-TGrou---0d2c9ac4\bin\testPkg\projects\System.Composition.TypedParts\netstandard1.7\project.csproj] [D:\j\workspace\windows-TGrou---0d2c9ac4\pkg\test\testPackages.proj]
D:\j\workspace\windows-TGrou---0d2c9ac4\bin\testPkg\tools\sdk\2.1.300-preview2-008530\Sdks\Microsoft.NET.Sdk\targets\Microsoft.PackageDependencyResolution.targets(205,5): error MSB4018: at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute() [D:\j\workspace\windows-TGrou---0d2c9ac4\bin\testPkg\projects\System.Composition.TypedParts\netstandard1.7\project.csproj] [D:\j\workspace\windows-TGrou---0d2c9ac4\pkg\test\testPackages.proj]
D:\j\workspace\windows-TGrou---0d2c9ac4\bin\testPkg\tools\sdk\2.1.300-preview2-008530\Sdks\Microsoft.NET.Sdk\targets\Microsoft.PackageDependencyResolution.targets(205,5): error MSB4018: at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask(ITaskExecutionHost taskExecutionHost, TaskLoggingContext taskLoggingContext, TaskHost taskHost, ItemBucket bucket, TaskExecutionMode howToExecuteTask) [D:\j\workspace\windows-TGrou---0d2c9ac4\bin\testPkg\projects\System.Composition.TypedParts\netstandard1.7\project.csproj] [D:\j\workspace\windows-TGrou---0d2c9ac4\pkg\test\testPackages.proj]
D:\j\workspace\windows-TGrou---0d2c9ac4\pkg\test\testPackages.proj(177,5): error MSB3073: The command "D:\j\workspace\windows-TGrou---0d2c9ac4\bin/testPkg/tools//dotnet msbuild /t:Test "D:\j\workspace\windows-TGrou---0d2c9ac4\bin/testPkg/test.msbuild" /p:TestPackages=System.Composition.TypedParts" exited with code 1.
```

See https://github.com/dotnet/corefx/issues/29341

/cc @nguerrera

**Comment by livarcocc** [comment]
We have fixed this one already at the tip of release/2.1.3xx. If you pick up the latest RC build, we added some resiliency to that operation there.

**Comment by ericstj** [comment]
Cool, thanks!

Fixed with https://github.com/dotnet/sdk/pull/2159
